presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -91,17 +94,12 @@ public QueryInfo(
         this.resetSessionProperties = ImmutableSet.copyOf(resetSessionProperties);
         this.outputStage = outputStage;
         this.failureInfo = failureInfo;
+        this.errorType = errorCode != null ? StandardErrorCode.toErrorType(errorCode.getCode()) : null;
```

Comment:
This would be a bit cleaner like this (flip the condition and static import):

```java
        this.errorType = (errorCode == null) ? null : toErrorType(errorCode.getCode());